### PR TITLE
Bluetooth: controller: Fix assert during DLE

### PR DESCRIPTION
**Pull request by cvinayak**
Fix assert in controller checking ticker_id_prepare = 0 when
a scanner and connection are active, and the connection is
skipping events in order to resize Rx buffers.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Comment by cvinayak** [comment]
@alexsven @joerchan this is a very obvious bug fix. Early code has LL_ASSERT in this if clause, now when making the events to skip till the pool is free for resizing needs the ticker_id_prepare to be reset. Please approve so that the PR is merged.

@aescolar please do the needful in the absence of @carlescufi

**Review by alexsven — APPROVED** [review]

**Review by joerchan — CHANGES_REQUESTED** [review]
Not sure how obvious that bug fix was. I think it would be easier to move the resetting to the early returns from `event_connection_prepare`.

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 1]
```diff
@@ -8082,6 +8082,10 @@ static inline int event_len_prep(struct connection *conn)
 			 * check in the next event if resize can be
 			 * done.
 			 */
+
+			/* Reset ticker_id_prepare as role is not continued */
+			_radio.ticker_id_prepare = 0U;
```
[the remark this answers is not in the document]
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ctrl.c]: agree about moving it to `event_connection_prepare`.  in the case of `event_conn_upd_prep`, on success the prepare is skip to be done later at the new win_offset.

**Review by joerchan — APPROVED** [review]
Consider making `event_conn_upd_prep` consistent with the `event_len_prep`.

**Pull request merged**